pkg/pullsync/pullsync_test.go

```diff
@@ -217,6 +217,20 @@ func haveChunks(t *testing.T, s *mock.PullStorage, addrs ...swarm.Address) {
 
 func newPullSync(s p2p.Streamer, o ...mock.Option) (*pullsync.Syncer, *mock.PullStorage) {
 	storage := mock.NewPullStorage(o...)
+	c := make(chan swarm.Chunk)
+	validator := &mockValidator{c}
 	logger := logging.New(ioutil.Discard, 0)
-	return pullsync.New(s, storage, logger), storage
+	return pullsync.New(s, storage, validator, logger), storage
+}
+
+type mockValidator struct {
+	c chan swarm.Chunk
+}
+
+func (*mockValidator) Validate(swarm.Chunk) bool {
+	return true
+}
+
+func (mv *mockValidator) ValidWithCallback(c swarm.Chunk) (bool, func()) {
+	return true, func() { mv.c <- c }
```

Comment:
is this channel used somehow? not sure why not to return an empty function here

Reply:
so that we test that it was called just like in your code https://github.com/ethersphere/bee/blob/master/pkg/pushsync/pushsync_test.go#L234

Reply:
i see you are passing this mock in the tests, but i can't see where do you actually assert that the call was made. this infers changes to the actual tests (which was not done). also, that is not my code :)